src/openrct2/object.h

```diff
@@ -65,7 +65,10 @@ typedef enum
  * size: 0x10
  */
 typedef struct rct_object_entry {
-    uint32 flags;
+    union {
+        uint8 end_flag; // needed not to read past allocated buffer.
```

Comment:
Why did you introduce this? Below you're still &-ing with 0xFF, so it doesn't seem to change anything.